scanner/src/main/kotlin/provenance/ProvenanceDownloader.kt

```diff
@@ -61,7 +61,7 @@ fun interface ProvenanceDownloader {
         // Use the provenanceDownloader to download each provenance from nestedProvenance separately, because they are
         // likely already cached if a path scanner wrapper is used.
 
-        val root = download(nestedProvenance.root)
+        val root = download(nestedProvenance.root as RemoteProvenance)
```

Comment:
I'm not yet convinced about the casts here an in other places, but I current do not have the time to think about it more.

Reply:
I understand your hesitation. I was not 100% happy with the change either.

I just added a commit 4b979057c04ba87245396ec2f78fae4612d76b68 to this PR, which removes the casts.
Unfortunately it also requires a lot of file changes.

Let me know which direction you prefer. Then I can either squash or remove the commit.

Reply:
Hi @sschuberth. Any way we can move forward on this?

@oss-review-toolkit/kotlin-devs Maybe someone else could have a look, if you are busy.

Reply:
Thanks for the reminder. I'm reviewing it now as we speak.